Engine/ac/overlay.cpp

```diff
@@ -390,24 +389,24 @@ void remove_screen_overlay(int type)
     dispose_overlay(over);
 
     // Don't erase vector elements, instead set invalid and record free index
-    screenover[type] = ScreenOverlay();
-    if (type >= OVER_FIRSTFREE)
-        over_free_ids.push(type);
+    screenover[id] = ScreenOverlay();
+    if (id >= OVER_FIRSTFREE)
+        over_free_ids.push(id);
 }
 
 void remove_all_overlays()
 {
-    for (int i = 0; i < screenover.size(); ++i)
+    for (uint32_t i = 0; i < screenover.size(); ++i)
         remove_screen_overlay(i);
 }
 
-ScreenOverlay *get_overlay(int index)
+ScreenOverlay *get_overlay(uint32_t id)
 {
-    assert(index >= 0 && index < screenover.size());
-    return (index >= 0 && index < screenover.size()) ? &screenover[index] : nullptr;
+    return (id < screenover.size() && screenover[id].id != ScreenOverlay::NoID)
```

Comment:
Hey, if id is lower than screenover.size(), isn't it automatically also not `ScreenOverlay::NoID` ? This is more a question really.